### PR TITLE
Don't show warnings for quotes during environment inspection

### PR DESCRIPTION
**Pull request by anderbubble**
`Executable` looks for arguments that contain quotes and issues a warning if found; but for some internal cases (e.g., running an embedded script with `-c`) quotes are necessary, but still generate warning messages to the end-user.

This adds a `_warn_quotes` kwarg to `Executable.__call__` so that the caller can intentionally opt-out of warnings.

### CODE REVIEW

File: lib/spack/spack/util/executable.py
Comment (adamjstewart) [inline lib/spack/spack/util/executable.py]: I would make this public, so use `warn_quotes` and mention it in the docstring.

Comment (anderbubble) [reply lib/spack/spack/util/executable.py]: Done.

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
Flake8:
```
lib/spack/spack/util/executable.py:147: [E501] line too long (83 > 79 characters)
lib/spack/spack/util/executable.py:152: [E501] line too long (80 > 79 characters)
```

**Comment by anderbubble** [comment]
@adamjstewart I fixed the flake8 errors. I used implicit string concatenation for one of the line lengths, which isn't generally a thing that I do; but it was the solution already used earlier in the function call, so I guess it's normal enough.

**Comment by anderbubble** [comment]
@adamjstewart anything remaining to allow this to be merged?

**Comment by adamjstewart** [comment]
I don't usually merge core-changes without multiple core developer approval. Maybe @scheibelp or @alalazo can take a look?

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this.

**Pull request closed without merging**
